core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -790,12 +790,16 @@ object IO {
   def reduceAll[E, A](a: IO[E, A], as: TraversableOnce[IO[E, A]])(f: (A, A) => A): IO[E, A] =
     as.foldLeft(a) { case (l, r) => l.par(r).map(o => f(o._1, o._2)) }
 
+  /**
+   * Merges a list of IO to a single IO, works in parallel.
+   */
+  def mergeAll[E, A, B](in: TraversableOnce[IO[E, A]])(zero: B, f: (B, A) => B): IO[E, B] =
+    in.foldLeft(IO.point[E, B](zero))((acc, a) => acc.par(a).map(o => f(o._1, o._2)))
```

Comment:
Wouldn't `f.tupled` be a tiny bit faster?

Reply:
Because it's generating less lambdas?